smartsim/_core/_install/builder.py

```diff
@@ -795,50 +795,90 @@ def __place_for_rai__(self, target: _PathLike) -> Path:
 
 
 class _WebArchive(_WebLocation):
+    """Used to download a remote resource"""
+
+    def __init__(self, headers: t.Optional[t.Dict[str, str]] = None) -> None:
+        """Initialize the instance"""
+        self._headers: t.Dict[str, str] = headers or {}
+
     @property
     def name(self) -> str:
-        _, name = self.url.rsplit("/", 1)
+        """Return the resource name identified by the URL."""
+        # omit the querystring to find the resource name
+        addressparts = self.url.split("?", maxsplit=1)
+        address = addressparts[0]
+        _, name = address.rsplit("/", 1)
         return name
 
     def download(self, target: _PathLike) -> Path:
+        """Retrieve the remote file
+
+        :param target: The desired target path for writing the downloaded file
+        :returns: If the file can be successfully downloaded, the path to the
+        downloaded file. Otherwise, None"""
```

Comment:
From the code and the function signature, it looks like we always return a `Path`, and never `None`.

Reply:
fixed! good eye.